ui/src/app/pages/app/component.ts

```diff
@@ -177,6 +267,12 @@ export class AppComponent implements OnInit {
     });
   }
 
+  ngOnDestroy() {
+    for (const s of this.subscriptions) {
```

Comment:
nit: should probably be consistent with other file in terms of `for` loops vs `foreach` on unsubscribe